### DISCUSSION TITLE
Setting timezone for calendar

### DISCUSSION THREAD

**Discussion by iqfx** [spatie/icalendar-generator#116] [Bugs]
Perhaps I don't quite understand how ICAL works, but when I create  a calendar using this code

```
$calendar = Calendar::create('test')->timezone(Timezone::create('Europe/Amsterdam'));

        // Add events to the calendar
        $events = Product::where('startDate', "!=", null)->get();

        foreach ($events as $event) {
            $calendar->event($this->createICalEvent($event));
        }

        return response($calendar->get())
            ->header('Content-Type', 'text/calendar');
```
returns this:

```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:spatie/icalendar-generator
NAME:test
X-WR-CALNAME:test
BEGIN:VTIMEZONE
TZID:Europe/Amsterdam
BEGIN:DAYLIGHT
DTSTART:20230326T020000Z
TZOFFSETFROM:+0100
TZOFFSETTO:+0200
END:DAYLIGHT
BEGIN:STANDARD
DTSTART:20231029T030000Z
TZOFFSETFROM:+0200
TZOFFSETTO:+0100
END:STANDARD
BEGIN:DAYLIGHT
DTSTART:20240331T020000Z
TZOFFSETFROM:+0100
TZOFFSETTO:+0200
END:DAYLIGHT
END:VTIMEZONE
BEGIN:VTIMEZONE
TZID:UTC
BEGIN:STANDARD
DTSTART:20230325T205339Z
TZOFFSETFROM:+0000
TZOFFSETTO:+0000
END:STANDARD
END:VTIMEZONE
BEGIN:VEVENT
UID:65835453db7ba
DTSTAMP:20231220T205339Z
```

How is it that TZID appears TWO times? One with Europe/Amsterdam, the second one being UTC?
I know it tries to keep in to account multiple timezones for multiple events (or however that works) But when I make sure all my dates are in Europe/Amsterdam timezone I just do not get why this library still thinks I want UTC. This anomaly mainly confuses google calendar and causes it to shows the wrong time (by adding an hour, I also checked the timezone of the calendar in google calendar and yup, it's GMT+0 while it should be GMT+1). Other calendars like gnome calendar do interpret the timezone correctly however.

Is there anyone that is familiar with this issue or can explain it to me better?

**Comment by rubenvanassche** [comment]
This is due to DTSTAMP which should always be in UTC, the package will generate a timezone for it. Not sure if that's a requirement from the spect though. Since we haven't heard from anyone else about this I don't think this is that big of an issue.

If you find some documentation that says that in this case no UTC timezone component is required and provide a PR that removes it then I'll gladly merge it.
